internal/cluster/refresh_status_preprocessor_test.go

```diff
@@ -0,0 +1,156 @@
+	"github.com/openshift/assisted-service/internal/host"
+	"github.com/openshift/assisted-service/internal/host/hostutil"
+	"github.com/openshift/assisted-service/internal/operators"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	"gorm.io/gorm"
+)
+
+var _ = Describe("Cluster Refresh Status Preprocessor", func() {
+
+	var (
+		preprocessor        *refreshPreprocessor
+		ctrl                *gomock.Controller
+		mockHostApi         *host.MockAPI
+		mockOperatorManager *operators.MockAPI
+		db                  *gorm.DB
+		dbName              string
+		cluster             *common.Cluster
+		clusterID           strfmt.UUID
+		ctx                 context.Context
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		ctrl = gomock.NewController(GinkgoT())
+		mockHostApi = host.NewMockAPI(ctrl)
+		mockOperatorManager = operators.NewMockAPI(ctrl)
+		mockOperatorManager.EXPECT().ValidateCluster(ctx, gomock.Any())
+		db, dbName = common.PrepareTestDB()
+		preprocessor = newRefreshPreprocessor(
+			logrus.New(),
+			mockHostApi,
+			mockOperatorManager,
+		)
+	})
+
+	AfterEach(func() {
+		common.DeleteTestDB(db, dbName)
+	})
+
+	createCluster := func() {
+		clusterID = strfmt.UUID(uuid.New().String())
+		testCluster := hostutil.GenerateTestCluster(clusterID)
+		cluster = &testCluster
+		clusterStatus := "insufficient"
+		cluster.Status = &clusterStatus
+		cluster.InstallationPreparationCompletionStatus = common.InstallationPreparationFailed
+		Expect(db.Create(&cluster).Error).ShouldNot(HaveOccurred())
+	}
+
+	deleteCluster := func() {
+		Expect(db.Delete(&common.Cluster{}, clusterID).Error).NotTo(HaveOccurred())
+	}
+
+	mockFailingValidator := func(context *clusterPreprocessContext) (ValidationStatus, string) {
+		return ValidationFailure, "Mock of a failed validation"
+	}
+
+	mockFailingCondition := func(context *clusterPreprocessContext) bool {
+		return false
+	}
+
+	mockFailAllValidations := func() {
+		for i, validation := range preprocessor.validations {
+			validation.condition = mockFailingValidator
+			preprocessor.validations[i] = validation
+		}
+
+		for i, condition := range preprocessor.conditions {
+			condition.fn = mockFailingCondition
+			preprocessor.conditions[i] = condition
+		}
+	}
+
+	Context("Skipping Validations", func() {
+
+		cantBeIgnored := common.NonIgnorableClusterValidations
+
+		var (
+			validationContext *clusterPreprocessContext
+		)
+
+		var conditionIsValidation = func(r *refreshPreprocessor, condition string) bool {
+			for _, validation := range r.validations {
+				if condition == validation.id.String() {
+					return true
+				}
+			}
+			return false
+		}
+
+		BeforeEach(func() {
+			createCluster()
+			mockFailAllValidations()
+			validationContext = newClusterValidationContext(cluster, db)
+		})
+
+		AfterEach(func() {
+			deleteCluster()
+		})
+
+		It("Should not ignore any validations if IgnoredClusterValidations is empty or invalid", func() {
```

Comment:
You check here only invalid, can we check nil too?

Reply:
No need to check for nil as a string cannot possibly have this value.